app/upgrades/xenon/upgrade.go

```diff
@@ -0,0 +1,86 @@
+package xenon
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	florinkeeper "github.com/noble-assets/florin/x/florin/keeper"
+	halokeeper "github.com/noble-assets/halo/x/halo/keeper"
+)
+
+var (
+	// TODO: Verify denom metadata
+
+	eureMetaData = banktypes.Metadata{
+		Description: "Regulated Euro Stablecoin",
+		DenomUnits: []*banktypes.DenomUnit{
+			{
+				Denom:    "aeure",
+				Exponent: 0,
+				Aliases:  []string{"attoeure"},
+			},
+			{
+				Denom:    "eure",
+				Exponent: 18,
+			},
+		},
+		Base:    "aeure",
+		Display: "eure",
+		Name:    "Euro Stablecoin",
+		Symbol:  "EURe",
+	}
+
+	usycMetaData = banktypes.Metadata{
+		Description: "",
+		DenomUnits: []*banktypes.DenomUnit{
+			{
+				Denom:    "uusyc",
+				Exponent: 0,
+				Aliases:  []string{"microusyc"},
+			},
+			{
+				Denom:    "usyc",
+				Exponent: 6,
+			},
+		},
+		Base:    "uusyc",
+		Display: "usyc",
+		Name:    "",
+		Symbol:  "USYC",
+	}
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	cfg module.Configurator,
+	halokeeper *halokeeper.Keeper,
+	florinkeeper *florinkeeper.Keeper,
```

Comment:
Let's make sure to not override the imports here ...

```suggestion
	haloKeeper *halokeeper.Keeper,
	florinKeeper *florinkeeper.Keeper,
```

Reply:
Good eye